tests/sles4sap/bone_install.pm

```diff
@@ -0,0 +1,76 @@
+# SUSE's SLES4SAP openQA tests
+#
+# Copyright 2019-2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Install SAP Business One via command line.
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+use base 'sles4sap';
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use Utils::Backends;
+use utils qw(file_content_replace zypper_call);
+use Utils::Systemd 'systemctl';
+use version_utils 'is_sle';
+use POSIX 'ceil';
+use Utils::Logging 'save_and_upload_log';
+
+sub run {
+    my ($self) = @_;
+    my ($proto, $path) = $self->fix_path(get_required_var('BONE'));
+    my $sid = get_required_var('INSTANCE_SID');
+    my $instid = get_required_var('INSTANCE_ID');
+    # set timeout as 4800 as a temp workaround for slow nfs
+    my $tout = get_var('HANA_INSTALLATION_TIMEOUT', 4800);    # Timeout for HANA installation commands.
+
+    select_serial_terminal;
+    #select_console 'x11';
+
+    # Transfer media.
+    my $target = '/sapinst';    # Directory in SUT where install media will be copied token
+    if (get_var 'ASSET_0') {
+        # If the ASSET_0 variable is defined, the test will attempt to download
+        # the HANA media from the factory/other directory of the openQA server.
+        record_info "Dowloading using ASSET_0";
+        download_hana_assets_from_server(target => $target, nettout => $tout);
+    }
+    elsif (get_required_var 'BONE') {
+        # If not, the media will be retrieved from a remote server.
+        record_info "Downloading using $proto";
+        $self->copy_media($proto, $path, $tout, $target);
+    }
+
+    # install bin is used for installation, verify if it exists
+    my $install_bin = '/sapinst/' . get_var('B1_INSTALL', "Packages.Linux/ServerComponents" . "/install");
+    die "install is not in [$install_bin]. Set B1_INSTALL to the appropiate relative path. Example: Packages.Linux/ServerComponents/install" if (script_run "ls $install_bin");
+
+    # Install B1
+    # Download xml config
+    my $b1_cfg = "bone.cfg";
+    my $hostname = script_output 'hostname';
+    my $local_url = "http://10.100.103.247:8000/";
+    my $admin_id = "ndbadm";
+
+    assert_script_run "curl -f -v " . $local_url . "$b1_cfg -o /tmp/$b1_cfg";
+    #assert_script_run "curl -f -v " . autoinst_url . "/data/sles4sap/$b1_cfg -o /tmp/$b1_cfg";
+
+    # change some default values
+    file_content_replace("/tmp/$b1_cfg", '%SERVER%' => $hostname);
+    file_content_replace("/tmp/$b1_cfg", '%INSTANCE%' => $instid);
+    file_content_replace("/tmp/$b1_cfg", '%TENANT_DB%' => $sid);
+    file_content_replace("/tmp/$b1_cfg", '%PASSWORD%' => $sles4sap::instance_password);
+    file_content_replace("/tmp/$b1_cfg", '%ADMIN_ID%' => $admin_id);
```

Comment:
I believe there's away to combine all `file_content_replace` calls into one. Something like:
```
file_content_replace("/tmp/$b1_cfg", '%SERVER%' => $hostname, '%INSTANCE%' => $instid, '%TENANT_DB%' => $sid, '%PASSWORD%' => $sles4sap::instance_password, '%ADMIN_ID%' => $admin_id);
```